### PR TITLE
Feat/add to cart

### PR DESCRIPTION
**Pull request by itsDiwaker**
@CristinaSolana it's not ready for development branch yet. I have created this pull request so that you can review and give your early feedback.

### CODE REVIEW

**Comment by itsDiwaker** [comment]
FYI @CristinaSolana 

**Few notes about this pull request:**

- Ajax activity loader has been implemented. It's available to all pages on `window.activityLoader` property.

- **AddToCart** plugin needs to called on pages seperately where add to cart button is available. Add to Cart button must have class `js-btn-add-to-cart` and `data-product-sku="sku"`

**TODO items:**

- Product page change `data-product-sku` attribute value based on the selected `color` and `size`. Those functionalities are yet to be implemented.

- Error handling like 404, 500 status codes handling with proper alerts.

**Comment by CristinaSolana** [comment]
cart lines failure...
screen shot 2017-09-25 at 9 40 20 am

**Comment by itsDiwaker** [comment]
<em>
This request should probably not be made individually on each page, but in the template itself where the header resides? Is there a reason it would be needed on individual pages?
</em>

Not all pages will have add-to-cart buttons that's why it's currently being called on each page separately. We can call it on `template.liquid` or `footer_script.liquid`, that will work too.

**Comment by itsDiwaker** [comment]
Reported issues have been fixed. We are still receiving 404 for `api/cart/addLine` due to ongoing work in backend.

**Comment by CristinaSolana** [comment]
ReferenceError bug
screen shot 2017-09-25 at 2 02 58 pm

**Comment by CristinaSolana** [comment]
@itsDiwaker Not all pages will have add to cart buttons, but all pages will need to get cart lines for the header.

File: assets/js/add-to-cart.js [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+                .innerHTML = cartItems.length;
+
+            var lineItemTemplateDomObj =  cartOverviewDomObj.body.querySelector(this.options.lineItemWrapperClassSelector + ' li');
+
+            //TODO: show real total when available in API response
+            var cartTotal = 0;
+
+            var lineItems = document.createDocumentFragment();
+
+            for(i = 0; i < cartItems.length; i++) {
```
Comment (CristinaSolana) [inline assets/js/add-to-cart.js]: i not declared as var

Comment (CristinaSolana) [inline assets/js/add-to-cart.js]: i not declared as var

File: templates/index.liquid [hunk 2]
```diff
@@ -712,11 +714,65 @@
...
+
+        {% if websiteSettings.pickGift.enabled and templateSettings.pag_custom_popup_enabled %}
+        $('#pag-custom-message').foundation('open');
+        {% endif %}
+
+        var addToCart = new AddToCart({
+            'ajaxService' : superagent,
+            'domParser': new DOMParser(),
+            'currencyMode' : '{{websiteSettings.storefrontPayments.uom}}',
+            'decimalPrecision' : '{{websiteSettings.storefrontPayments.pointConversion}}',
```
Comment (CristinaSolana) [inline templates/index.liquid]: decimal precision should be 2. websiteSettings.storefrontPayments.pointConversion is not related to this setting in any way.

File: assets/js/add-to-cart.js [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+                var lineItemNode = lineItemTemplateDomObj.cloneNode(true);
+
+                var lineItemData = cartItems[i];
+
+                var lineItemImg = lineItemData.item.imageUrl;
```
Comment (CristinaSolana) [inline assets/js/add-to-cart.js]: I updated name, image and price with new keys based on changes that were made on endpoint.

**Comment by CristinaSolana** [comment]
@itsDiwaker also,
- Issue with window resize and multiple calls persists on product.liquid view.
- adding to cart broken (let me know if this is related somehow to the api changes that were made)
screen shot 2017-09-25 at 3 02 13 pm

**Comment by itsDiwaker** [comment]
@CristinaSolana 

> ReferenceError bug
> screen shot 2017-09-25 at 2 02 58 pm

**Reply:** Must be happening in some specific scenario.  Can you share a test-case?

**Comment by itsDiwaker** [comment]
@CristinaSolana 

> @itsDiwaker Not all pages will have add to cart buttons, but all pages will need to get cart lines for the header.
**Reply:** Sorry that was stupid. I guess it stemmed from me thinking to have header update and add to cart as two separate plugins. Will make changes.

File: assets/js/add-to-cart.js (hunk 3 above)
Comment (itsDiwaker) [reply assets/js/add-to-cart.js]: thanks!

File: assets/js/add-to-cart.js (hunk 1 above)
Comment (itsDiwaker) [reply assets/js/add-to-cart.js]: updated!

File: templates/index.liquid (hunk 2 above)
Comment (itsDiwaker) [reply templates/index.liquid]: passed a hardcoded value 2 for now.

**Comment by itsDiwaker** [comment]
> @itsDiwaker also,
> - we should not show the loader for when getting line items for header cart list. Only when posting and putting.

**Reply:** done!

**Comment by CristinaSolana** [comment]
Fix home and product detail page to allow adding to cart, currently payload is empty object in array, should include skuCode and value from data-product-sku attr

**Pull request merged**